### PR TITLE
Move save_weights_to_checkpoint to shared utils

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

Move `save_weights_to_checkpoint` and rename/move `shard_checkpoint -> shard_jax_weights` in `checkpoint_conversion.utils.utils.py`, and update all affected dependencies ([10 files](https://screenshot.googleplex.com/BYCMGeGsHz48z68) have all been covered).

Now these functions can be easily shared among other scripts.

No any other changes rather than rename/move ([diff](https://diff.googleplex.com/#key=wnFCY89WDvF2))

# Tests

```
# to maxtext conversion
python -m maxtext.checkpoint_conversion.to_maxtext src/maxtext/configs/base.yml model_name=qwen3-4b base_output_directory=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-4b/scanned/2026-05-12 scan_layers=true hf_access_token=<your_token> weight_dtype=bfloat16 hardware=cpu skip_jax_distributed_system=True checkpoint_storage_use_ocdbt=False checkpoint_storage_use_zarr3=False --eager_load_method=safetensors

# logits check
python -m tests.utils.forward_pass_logit_checker maxtext/configs/base.yml run_name=ht_test model_name=qwen3-4b tokenizer_path=Qwen/Qwen3-4B load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-4b/scanned/2026-05-14/0/items scan_layers=True max_prefill_predict_length=4 ici_tensor_parallelism=4 hf_access_token=<your_token> --run_hf_model=True --hf_model_path=Qwen/Qwen3-4B --max_kl_div=0.015
```

[Logs](https://paste.googleplex.com/5569316285186048)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by shuningjin — APPROVED** [review]
Thanks for refactor! LGTM

**Review by SurbhiJainUSC — APPROVED** [review]
